src/Integration.Vsix/Telemetry/TelemetryLogger.cs

```diff
@@ -15,8 +16,8 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     [Export(typeof(ITelemetryLogger)), PartCreationPolicy(CreationPolicy.Shared)]
     internal class TelemetryLogger : ITelemetryLogger
     {
-        private readonly System.IServiceProvider serviceProvider;
-        private readonly SonarLintSqmCommandTarget sqmCommandHandler;
+        private readonly IServiceProvider serviceProvider;
+        private readonly Dictionary<TelemetryEvent, Action> eventLoggerMap = new Dictionary<TelemetryEvent, Action>();
```

Comment:
There was a cyclomatic complexity in the previous implementation, changing the code to use this dictionary instead of a switch solved that issue.
I've considered generating this code but wasn't trivial since we will need to generated the TelemetryEvent enum which is defined in a different project.